docs/DeveloperGuide.md

```diff
@@ -1,34 +1,179 @@
+|---------------------------|---------------------------------------------------------------|
+| EventParser               | EventCommandInterpreter                                       | 
+| CalenderParser            | EventCommandInterpreter                                       | 
+| AttendanceParser          | Step-by-step command at Attendance-related command classes    |
+| PerformanceParser         | Step-by-step command at performance-related command classes   |  
+
+A Parser class is created when a user input contains data to be stored or used in certain features.    
 
-## Design & Implementation
+### UI
+![Ui](images/Ui.png "Class diagram of Ui component")                
+*Class diagram of the UI component*   
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+UI is the main class handles user display, which includes reading user input and printing information 
+back to the user on command-line.  
+Besides the normal command line messages, there are two subclasses of UI: 
+DisplayList and DisplayTable, to specifically print the list and table interface to professor. 
 
+## Feature Design and Implementation 
+### Event
+![event](images/event.png "Class diagram of Event component")           
+*Class diagram of the Event component*
+
+1. When a user enters an event-related command, the command is analysed by `EventCommandInterpreter`. 
+1. Once determined, the relevant information (e.g. index, name, time, date, venue) are extracted by 
+`EventParser`.
+1. Then, the relevant class that corresponds to the command is created, with the information extracted 
+from the previous step passed into it. It modifies `Event` or `EventList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
+Note that:
+* `datetime` is stored as a single attribute in `Event` class, but it is exposed to user as `date` 
+and `time`, which corresponds to `d/` and `t/` flag respectively.
+* `editDate` or `editTime` commands are not available. Only `editDateTime` is available to change the 
+`date` and/or `time` of an `Event` object.
+* `delete(Event)` method is currently not in use, but can be used to implement delete by event name, 
+either by complete match, or fuzzy match.
+* Any classes (e.g. `Seminar`) that inherit from `Event` class will have similar control flow. 
+
+### Attendance
+![attendance](images/Attendance.png "Class diagram of Attendance component")        
+*Class diagram of the Attendance component*
+1. When a user enters an attendance-related command, the command is analysed by `AttendanceCommandInterpreter`. 
+1. Once determined, the relevant class that corresponds to the type of command is created.
+1. Then, the class will execute base on its function. It modifies `AttendanceList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
+###Calendar Implementation
```

Comment:
![image](https://user-images.githubusercontent.com/49228945/77993506-e08fe900-735a-11ea-86b1-22877d0e8f8c.png)
It seems that the title here is not working well. Maybe consider to add one more space before "Calender Implementation"? i.e. `### Calendar Implementation` instead of "###Calendar Implementation"